client/my-sites/domains/domain-management/list/domain-row.jsx

```diff
@@ -98,22 +98,26 @@ class DomainRow extends PureComponent {
 	}
 
 	renderSite() {
-		const { site } = this.props;
+		const { domain } = this.props;
+		if ( domain.isDomainOnlySite ) {
+			return null;
+		}
+
 		return (
 			<div className="domain-row__site-cell">
-				<Button href={ '/home/' + site?.slug } plain>
-					{ ! site.options?.is_domain_only ? site?.title || site?.slug : '' }
+				<Button href={ '/home/' + domain.blogId } plain>
+					{ ( domain.siteTitle || domain.siteSlug ) ?? '' }
```

Comment:
Is that the same thing?

```suggestion
					{ domain.siteTitle ?? domain.siteSlug ?? '' }
```